lib/CL/devices/basic/basic.c

```diff
@@ -222,11 +222,17 @@ pocl_basic_init (unsigned j, cl_device_id device, const char* parameters)
 void
 pocl_basic_run (void *data, _cl_command_node *cmd)
 {
+  cl_kernel kernel = cmd->command.run.kernel;
+  if (kernel->custom_runner)
```

Comment:
This breaks the responsibilities a bit. Each device driver should instead implement the DBKs and at run we should choose the implementation at the driver side (based on DBK name).